ecl/eclagent/eclagent.cpp

```diff
@@ -1860,21 +1859,25 @@ void EclAgent::setRetcode(int code)
 
 void EclAgent::runWorkunitAnalyser(IWorkUnit * w, const char * optGraph)
 {
-    if (w->getDebugValueBool("analyzeWorkunit", agentTopology->getPropBool("@analyzeWorkunit", true)))
+    if (w->hasDebugValue("analyzeWorkunit") && !w->getDebugValueBool("analyzeWorkunit", true))
+        return;
+    if (!getBoolWUOption(nullptr, nullptr, "analyzerOptions/@disabled", false))
```

Comment:
Why not pass w and "analyzeWorkunit" to the function?

Reply:
Because if these are passed into the function and 'analyzeWorkunit' was true, then the getBoolWUOption would also return true and that would mean the workunit would _not_ be analyzed, which is not what it should do:

if analyzerOptions/@disabled == true, then it shouldn't execute the analysis.  However, if analyzeWorkunit == true, it should analyze the workunit.